src/main/java/com/github/pereiratostain/generator/MermaidSchemaGenerator.java

```diff
@@ -3,21 +3,30 @@
 import static java.util.Objects.requireNonNull;
 
 import com.github.forax.umldoc.core.Entity;
+import com.github.forax.umldoc.core.Field;
 import com.github.forax.umldoc.core.Modifier;
 import java.io.IOException;
 import java.io.Writer;
+import java.util.ArrayList;
+import java.util.HashSet;
 import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
 
 /**
  * Generate a schema in the Mermaid format.
  */
 public class MermaidSchemaGenerator implements Generator {
+  private HashSet<String> entitiesName;
```

Comment:
The class is not thread safe anymore ?
This should not be stored in a field given that those informations are only needed when calling generate() and useless after the call ends

Reply:
The field entitiesName is used in computeFieldsClass() and need to be first initialized in generate() to have all enitities.

Reply:
That's the issue, methods should be `pure` by default, i.e. have the result only depending of the value of the parameters.

So fix the issue by passing the set as parameter and the code will become thread-safe.

Reply:
Understood, done